### PR TITLE
Added `isAssignableFrom(Type)` to TypeChecker

### PR DESCRIPTION
**Pull request by kevmoo**
Building off https://github.com/dart-lang/source_gen/pull/153

### CODE REVIEW

**Comment by natebosch** [comment]
Commit messages aren't very descriptive...

**Comment by kevmoo** [comment]
@jakemac53 PTAL
@natebosch I didn't ask you 👅

**Comment by kevmoo** [comment]
@matanlurey FYI...

File: lib/src/json_serializable/type_helper.dart [hunk 1]
```diff
@@ -104,7 +105,7 @@ class DateTimeHelper extends TypeHelper {
   bool canDeserialize(DartType type) => _matchesType(type);
 
   bool _matchesType(DartType type) =>
-      type.element.library?.isDartCore == true && type.name == 'DateTime';
```
Comment (kevmoo) [inline lib/src/json_serializable/type_helper.dart]: @matanlurey Dude – this is **soooo** much better. I was looking into cleaning this up for a while. Love your solution...

**Pull request merged**